In [1]:
import torch
import torch.nn as nn
# Set random seeds
torch.manual_seed(42) 
torch.cuda.manual_seed(42)
torch.cuda.manual_seed_all(42)


from model import SimpleNet, ColorNet
from dataloader import Color_Dataloader
from train import train

import warnings
warnings.filterwarnings("ignore")

In [2]:
DIR_DATASET = 'dataset'
PATH_MODEL = 'weights/simple_model.pt'
NUM_EPOCHS = 50
NUM_CLASSES = 9
PATIENT = 5
LR = 0.001
BATCH_SIZE = 10
DEVICE = 'cuda'

In [3]:
train_loader, test_loader = Color_Dataloader(
    dir_dataset=DIR_DATASET,
    batch_size=BATCH_SIZE
)

In [4]:
net = ColorNet(
    num_classes = NUM_CLASSES
)
model = net.to(DEVICE)

In [5]:
loss_fn = nn.CrossEntropyLoss()
#optimizer = torch.optim.Adam(params=model.parameters(), lr=LR)
optimizer = torch.optim.SGD(params=model.parameters(), lr=LR, momentum=0.9)
model_0_results = train(model=model, 
                        train_dataloader=train_loader,
                        test_dataloader=train_loader,
                        optimizer=optimizer,
                        loss_fn=loss_fn, 
                        epochs=NUM_EPOCHS,
                        patient= PATIENT,
                        model_path=PATH_MODEL,
                        device=DEVICE)


  0%|          | 0/50 [00:00<?, ?it/s]

Model improve from  0.0000 to  0.1666
Model improve from  0.1666 to  0.3637
Model improve from  0.3637 to  0.4583
Model improve from  0.4583 to  0.5627
Model not improver 1/5
Model not improver 2/5
Model improve from  0.5627 to  0.6065
Model improve from  0.6065 to  0.6381
Model improve from  0.6381 to  0.6477
Model not improver 1/5
Model improve from  0.6477 to  0.7753
Model not improver 1/5
Model not improver 2/5
Model not improver 3/5
Model not improver 4/5
Model improve from  0.7753 to  0.8449
Model not improver 1/5
Model not improver 2/5
Model not improver 3/5
Model not improver 4/5
Model improve from  0.8449 to  0.8761
Model not improver 1/5
Model improve from  0.8761 to  0.8803
Model not improver 1/5
Model not improver 2/5
Model not improver 3/5
Model not improver 4/5
Model improve from  0.8803 to  0.9101
Model not improver 1/5
Model not improver 2/5
Model not improver 3/5
Model not improver 4/5
Model improve from  0.9101 to  0.9115
Model not improver 1/5
Model improve from  0.9

# Simple Net
 batch_size | dropout | learning rate |F1-train |F1-test|
| --- | --- | --- | --- | --- |
| 32 | 0.4 | 0.001| 0.88|0.79|
| 32 | 0.2 | 0.001| 0.82|0.74|
| 32 | 0.5 | 0.001| 0.81|0.76|
| 64 | 0.4 | 0.001| 0.80|0.67|
| 32 | 0.4 | 0.005| 0.76|0.48|
| 32 | 0.4 | 0.0005| 0.74|0.73|
| 32 | 0.4 | 0.0007| 0.88|0.79|
| 32 | 0.4 | 0.00085| 0.89|0.79|

# Color NET
# Simple Net
 batch_size |learning rate |F1-train |F1-test|
| --- | --- | --- | --- |
<!-- | 10 | 0.001| 0.88|0.79| -->
